gui/filebrowser-dialog.h

```diff
@@ -45,7 +45,7 @@ class FileBrowserDialog : public Dialog {
 
 	void handleCommand(CommandSender *sender, uint32 cmd, uint32 data) override;
 
-	const char *getResult() { return Dialog::getResult() ? _fileName->getEditString().c_str() : nullptr; }
+	const char *getResult() { return Dialog::getResult() ? _fileName->getEditString().encode().c_str() : nullptr; }
```

Comment:
You have another memory issue here.
`_fileName->getEditString()` returns a reference to to the string stored in the `_fileName` widget, so getting its `c_str()` is fine (as long as you don't use it after the widget has been deleted.
However the addition of `encode()` means it is now creating a temporary `Common::String` and you are returning a pointer to the content of this temporary string. And by the time the call of this function can use the pointer, the string will already have been destroyed.

Maybe it should return a `Common::String` instead.

Reply:
I checked for usage of this function
(Examples: `engines/director/lingo/xlibs/fileio.cpp:140`, `engines/bbvs/minigames/bbairguitar.cpp:1220:` 
and it seems like they use a `Common::String` for assignment anyway. So, making the return value a `const char *` is useless anyway, since it promotes to a `Common::String`. So, your suggestion is correct, and I will change this to return a string instead.